app/src/test/java/com/example/parkingsystem/ParkingReservationPresenterTest.java

```diff
@@ -0,0 +1,331 @@
+        long nowDate = new Date().getTime();
+
+        // When
+        presenter.validateDates(startDate, endDate);
+
+        // Then
+        Assert.assertTrue(startDate < endDate);
+        Assert.assertFalse(startDate > nowDate);
+        Assert.assertTrue(nowDate < endDate);
+        Assert.assertFalse(nowDate < startDate);
+
+        verify(view).showInconsistentDates();
+    }
+
+    @Test
+    public void validateDates_endDateIsInThePast_isFalse() {
+        //Given
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, 3);
+        Date dateForStart = calendar.getTime();
+        long startDate = dateForStart.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, -5);
+        Date dateForEnd = calendar.getTime();
+        long endDate = dateForEnd.getTime();
+        long nowDate = new Date().getTime();
+
+        // When
+        presenter.validateDates(startDate, endDate);
+
+        // Then
+        Assert.assertTrue(startDate > endDate);
+        Assert.assertTrue(startDate > nowDate);
+        Assert.assertFalse(endDate > nowDate);
+
+        verify(view).showInconsistentDates();
+    }
+
+    @Test
+    public void validateDates_isNull() {
+        // When
+        presenter.validateDates(0, 0);
+        // Then
+        verify(view).showEmptyDates();
+    }
+
+    @Test
+    public void validateSecurityCode_isTrue() {
+        //Given
+        String code = "ADN123";
+        when(view.getSecurityCode()).thenReturn(code);
+
+        //When
+        presenter.validateSecurityCode(code);
+
+        //Then
+        Assert.assertNotNull(code);
+    }
+
+    @Test
+    public void validateSecurityCode_codeIsEmpty_isFalse() {
+        //Given
+        String code = "";
+        when(view.getSecurityCode()).thenReturn(code);
+
+        //When
+        presenter.validateSecurityCode(code);
+
+        //Then
+        Assert.assertNotNull(code);
+        verify(view).showCodeNotComplaint();
+    }
+
+    @Test
+    public void validateSecurityCode_codesIsTooLarge_isFalse() {
+        //Given
+        String code = "ABCDEFGHIJKLMNOPQRSTUVWXYZ123456";
+        when(view.getSecurityCode()).thenReturn(code);
+
+        //When
+        presenter.validateSecurityCode(code);
+
+        //Then
+        Assert.assertNotNull(code);
+        verify(view).showCodeNotComplaint();
+    }
+
+    @Test
+    public void validateSecurityCode_isNull() {
+        presenter.validateSecurityCode(null);
+
+        verify(view).showCodeNotComplaint();
+    }
+
+    //ToDo check test for parkingLot greater than parkingSize
+    @Test
+    public void validateParkingLotNumber_isTrue() {
+        //Given
+        when(model.setParkingLotNumber(view.getParkingLotNumberEntered())).thenReturn(10);
+        when(model.getParkingSize()).thenReturn(20);
+
+        //When
+        presenter.validateParkingLotNumber(10);
+
+        //Then
+        Assert.assertTrue(presenter.validateParkingLotNumber(10));
```

Comment:
Instead of calling validateParkingLotNumber() twice. Store result in local variable.